File: codes/ComplexNumberMultiply537/Solution.java
package codes.ComplexNumberMultiply537;

public class Solution{
    public String complexNumberMultiply(String num1, String num2) {
        int[] number1 = getCount(num1);
        int[] number2 = getCount(num2);
        int a = number1[0]*number2[0] - number1[1]*number2[1];
        int b = number1[0]*number2[1] + number1[1]*number2[0];
        String ans = a + "+" + b + "i";
        return ans;
    }

    public int[] getCount(String num){
        num = num.substring(0,num.length()-1);
        String[] sNums = num.split("\\+");
        int[] nums = new int[sNums.length];
        for(int i = 0; i<sNums.length; i++){
            nums[i] = Integer.parseInt(sNums[i]);
        }
        return nums;
    }

    public static void main(String[] args) {
        String x1 = "1+1i";
        String x2 = "1+1i";
        Solution solution = new Solution();
        System.out.println(solution.complexNumberMultiply(x1,x2));
    }
}

File: codes/ComplexNumberMultiply537/solution.cs
public class Solution {
    public string ComplexNumberMultiply(string num1, string num2) {
        // 题目很简单，但是练习的api很多
        // 1. 字符串分割
        // 2. string和int转换
        // 3. 设定返回字符串的格式
        char[] splitChar = new char[]{'+', 'i'};
        string[] complex1 = num1.Split(splitChar);
        string[] complex2 = num2.Split(splitChar);
        int real1 = int.Parse(complex1[0]);
        int imaginary1 = int.Parse(complex1[1]);
        int real2 = int.Parse(complex2[0]);
        int imaginary2 = int.Parse(complex2[1]);

        return string.Format("{0}+{1}i", real1 * real2 - imaginary1 * imaginary2, real1 * imaginary2 + real2 * imaginary1);
    }
}